underfs/swift/src/main/java/alluxio/underfs/swift/SwiftUnderFileSystem.java

```diff
@@ -264,14 +270,13 @@ public long getSpace(String path, SpaceType type) throws IOException {
 
   @Override
   public boolean isFile(String path) throws IOException {
-    String strippedPath = stripPrefixIfPresent(path);
-    return mAccount.getContainer(mContainerName).getObject(strippedPath).exists();
+    return exists(path) && !isDirectory(path);
```

Comment:
FWIW, this is consistent with what we do for other object storage (GCS, S3, OSS). I think the motivation behind this is to optimize for the case where the path does not exist. However, as you correctly point out that it makes the case where the path exists more expensive. I think the right thing to do is to actually avoid reusing exists and isDirectory and instead write a custom function, which is what I will do.